compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1046,20 +1049,31 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
                                     continue;
                                 }
                                 this.insert_lifetime(lt, Region::Static);
-                                this.tcx
-                                    .sess
-                                    .struct_span_warn(
-                                        lifetime.span,
-                                        &format!(
-                                            "unnecessary lifetime parameter `{}`",
-                                            lifetime.name.ident(),
-                                        ),
-                                    )
-                                    .help(&format!(
-                                        "you can use the `'static` lifetime directly, in place of `{}`",
-                                        lifetime.name.ident(),
-                                    ))
-                                    .emit();
+
+                                let hir_map = this.tcx.hir();
+                                let parent = hir_map.get_parent_node(lt.hir_id);
+
+                                match hir_map.get(parent) {
+                                    hir::Node::Item(&hir::Item { kind: hir::ItemKind::OpaqueTy(..), .. }) => {
+                                        // do not warn
```

Comment:
Why do we need this?

Reply:
Because due to this `'static` lifetime trick we end with `'static` in the where clauses of the RPIT. This silences the warning for this case.

Reply:
Document this in the source